hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -24,116 +24,112 @@
       tests: ["flash_ctrl_smoke"]
     }
```

Comment:
one thing that is not explicitly listed..because flash buffers previously read data, we need to also handle the "eviction cases" if the buffered data becomes stale.

Ie, read read location A, and then program / erase location A and finally read A again.  This should cause the flash controller to evict the previously buffered data and retrieve the new data.

Reply:
another one worthy of a directed test is the different info partitions that have special properties associated with them based on external input, for example isolated partition.  I think this is indirectly mentioned in your tests, but I just want to make sure.

Reply:
the prim_flash is able to create errors when read also. This is not something directly under the control of the open source portion, so it would be good to emulate that error and ensure everything is behaving correctly.

Reply:
note also in the memory properties registers, there is also another attribute (high endurance).  From open source testing, this bit will not make a difference in function, but we need to make sure on the prim_flash boundary it sees the attribute bit if a given region has that attribute set.

Reply:
o i didn't see any specific test that targeted bank erase, there was a recently discovered bug in #8934

Reply:
there's also a register, erase suspend where i did not see an explicit test for.  This one is once again more of a connectivity check.

Reply:
the flash has an idle indication to the pwrmgr to inform it of any ongoing stateful activity (erase / program), it would be good to check for correct expectation / behavior there.

Reply:
related to ECC, there is a bit that controls whether ECC data errors return error or not.
This is mainly meant for software debugging purposes so taht we can observe the raw data in case of multi-bit errors, it would be good to make sure this is covered.

Reply:
there is currently a scratch register, but i can remove that as it serves no real purpose.

Reply:
> one thing that is not explicitly listed..because flash buffers previously read data, we need to also handle the "eviction cases" if the buffered data becomes stale.
> 
> Ie, read read location A, and then program / erase location A and finally read A again. This should cause the flash controller to evict the previously buffered data and retrieve the new data.

this can be added to flash_ctrl_read_direct_b2b test.

Reply:
> another one worthy of a directed test is the different info partitions that have special properties associated with them based on external input, for example isolated partition. I think this is indirectly mentioned in your tests, but I just want to make sure.

Test host_ctrl_hw_if is planned to cover reading of seed materials and RMA.

Reply:
> the prim_flash is able to create errors when read also. This is not something directly under the control of the open source portion, so it would be good to emulate that error and ensure everything is behaving correctly.

Test flash_ctrl_errors is meant for covering all errors listed in register model and here we could also add emulation of memory error.

Reply:
> note also in the memory properties registers, there is also another attribute (high endurance). From open source testing, this bit will not make a difference in function, but we need to make sure on the prim_flash boundary it sees the attribute bit if a given region has that attribute set.

This can be added into flash_ctrl_full_mem_access test.

Reply:
> o i didn't see any specific test that targeted bank erase, there was a recently discovered bug in #8934

Same as previous comment.

Reply:
> there's also a register, erase suspend where i did not see an explicit test for. This one is once again more of a connectivity check.

I assume that  erasing suspension is ongoing in Pyisical Controller while erase of Bank is ongoing (some pages are already erased and we want to stop further erase. If this feature is not complex it can be part flash_ctrl_host_arb. If not, we need separate test.

Reply:
> the flash has an idle indication to the pwrmgr to inform it of any ongoing stateful activity (erase / program), it would be good to check for correct expectation / behavior there.

From Flash CTRL HWIP Technical spec, i can see only pwrmgr_i signal. if this is only observing status of current operation via some power pins, it can be tested smoke_sw_op test.

Reply:
> related to ECC, there is a bit that controls whether ECC data errors return error or not. This is mainly meant for software debugging purposes so taht we can observe the raw data in case of multi-bit errors, it would be good to make sure this is covered.

 I assume ECC_EN_x bits are those. We can test this at flash_ctrl_ecc test.

Reply:
> there is currently a scratch register, but i can remove that as it serves no real purpose.

Ok, fine with me.

Reply:
Regarding the idle indication for pwrmgr, there is a pwrmgr test for this, but what we need is to make sure the idle output is internally correct. One way to do it could be to add an SVA for idle to be active only when there is no activity in flash_ctrl, or you could check this in the scoreboard.